source/data-modeling/inheritance.txt

```diff
@@ -0,0 +1,306 @@
+``discriminator_key()`` method.
+
+The following example demonstrates how to set a custom discriminator key
+when defining a model class:
+
+.. code-block:: ruby
+   :emphasize-lines: 6
+
+   class Person
+     include Mongoid::Document
+   
+     field :name, type: String
+
+     self.discriminator_key = "sub_type"
+   end
+
+When you create an instance of ``Person`` or any of its child classes,
+{+odm+} adds the ``sub_type`` field to documents in MongoDB.
+
+.. note::
+
+   You can change the discriminator key only on the parent class.
+   {+odm+} raises an error if you set a custom key on any child
+   class.
+
+If you change the discriminator key after defining a child class, {+odm+}
+adds the new key field, but the old field is unchanged. For example, suppose
+you add the following code to your application *after* defining your
+model classes:
+
+.. code-block:: ruby
+
+   Person.discriminator_key = "sub_type"
+
+In this case, when you create an instance of a child class such as
+``Employee``, {+odm+} adds both the ``sub_type`` and ``_type`` fields to
+the document.
+
+You can also change the discriminator key at the global level, so that
+all classes use the specified key instead of the ``_type`` field.
+
+You can set a global key by adding the following code to your
+application *before* defining any model classes:
+
+.. code-block:: ruby
+
+   Mongoid.discriminator_key = "sub_type"
+
+All classes use ``sub_type`` as the discriminator key and do not include
+the ``_type`` field.
+
+.. note::
+
+   You must set the discriminator key on the global level before defining
+   any child classes for the classes to use that global value. If you set
+   the global key after defining child classes, your saved documents
+   contain the default ``_type`` field.
+
+Change the Discriminator Value
+------------------------------
+
+You can customize the value that {+odm+} sets as the discriminator value
+in MongoDB. Use the ``discriminator_value()`` method when defining a
+class to customize the discriminator value, as shown in the following
+example:
+
+.. code-block:: ruby
+   :emphasize-lines: 6
+
+   class Employee
+     include Mongoid::Document
+   
+     field :company, type: String
+
+     self.discriminator_value = "Worker"
+   end
+
+When you create an instance of ``Employee``, the document's ``_type``
+discriminator field has a value of ``"Worker"`` instead of the class
+name.
+
+.. note::
+
+   Because the discriminator value customization is declared in child classes,
+   you must load the child classes retrieved by a query *before* sending
+   that query. In the preceding example, the ``Employee`` class definition
+   must be loaded before you query on ``Person`` if the returned documents could
+   potentially be instances of ``Employee``. Autoloading isn't able to resolve
+   the discriminator value ``"Worker"`` to return the document as an
+   instance of ``Employee``.
```

Comment:
[s] Can potentially omit extra words for brevity and make this note easier to follow, for example:
```suggestion
.. note::

   Because the discriminator value customization is declared in child classes,
   you must load the child classes retrieved by a query *before* sending
   that query. In the preceding example, the ``Employee`` class definition
   must be loaded before you query on ``Person`` if the returned documents could
   be instances of ``Employee``. Autoloading cannot resolve
   the discriminator value ``"Worker"`` to return the document as an
   instance of ``Employee``.
```